docs/pkg/client/events.md

```diff
@@ -0,0 +1,376 @@
+---
+title: EventsQueryClient Components
+---
+
+flowchart
+
+  subgraph comet[Cometbft Node]
+    subgraph rpc[JSON-RPC]
+      sub[subscribe endpoint]
+    end
+  end
+
+  subgraph eqc[EventsQueryClient]
+    q1_eb[EventsBytesObservable]
+    q1_conn[Connection]
+    q1_dial[Dialer]
+  end
+
+  q1_obsvr1[Observer 1]
+  q1_obsvr2[Observer 2]
+
+
+  q1_obsvr1 --"#Subscribe()"--> q1_eb
+  q1_obsvr2 --"#Subscribe()"--> q1_eb
+
+
+  q1_dial =="#DialContext()"==> q1_conn
+  q1_eb --"#Receive()"--> q1_conn
+
+  q1_conn -.-> sub
+```
+
+```mermaid
+---
+title: EventsReplayClient Components
+---
+
+flowchart
+  subgraph chain[Pocket Network]
+    c1_md[Action]
+  end
+
+  subgraph erc[EventsReplayClient]
+    subgraph eqc[EventsQueryClient]
+      r1_qs[Event Query String]
+      r1_qe[Websocket URL]
+    end
+    r1_rt[Publish Events Factory]
+    subgraph dec[EventBytesDecoder]
+      r1_df[Decoder Function]
+    end
+    r1_ob[LatestReplayObservable]
+    r1_pb[LatestReplayObservablePublisher]
+  end
+
+  subgraph con[Consumer]
+    c1_ob[Replay Observable]
+    c1_ev[Events]
+  end
+
+  c1_md -."#EmitTypedEvent".-> eqc
+
+  eqc --"retry.OnError"--> r1_rt
+  r1_df --"Decoded Event"--> r1_rt
+  r1_rt -."#EventBytes".-> eqc
+  r1_rt --"LatestReplayObservable"-->r1_pb
+  r1_pb --> r1_ob
+
+  eqc --"Serialised Event"--> r1_df
+  r1_ob --"#EventsSequence"--> c1_ob
+  r1_ob --"#LastNEvents"--> c1_ev
+```
+
+### Subscriptions
+
+```mermaid
+---
+title: Event Subscription Data Flow
+---
+
+flowchart
+
+  subgraph comet[Cometbft Node]
+      subgraph rpc[JSON-RPC]
+          sub[subscribe endpoint]
+      end
+  end
+
+  subgraph eqc[EventsQueryClient]
+      subgraph q1[Query 1]
+          q1_eb[EventsBytesObservable]
+          q1_conn[Connection]
+      end
+      subgraph q2[Query 2]
+          q2_conn[Connection]
+          q2_eb[EventsBytesObservable]
+      end
+  end
+
+  q1_obsvr1[Query 1 Observer 1]
+  q1_obsvr2[Query 1 Observer 2]
+  q2_obsvr[Query 2 Observer]
+
+  q1_eb -.-> q1_obsvr1
+  q1_eb -.-> q1_obsvr2
+  q2_eb -.-> q2_obsvr
+
+
+  q1_conn -.-> q1_eb
+  q2_conn -.-> q2_eb
+
+  sub -.-> q1_conn
+  sub -.-> q2_conn
+```
+
+## Installation
+
+```bash
+go get github.com/pokt-network/poktroll/pkg/client/events
+```
+
+## Features
+
+- **Websocket Connection**: Uses the
+  [Gorilla WebSockets](https://github.com/gorilla/websocket) for implementing the
+  connection interface.
+- **Events Subscription**: Subscribe to chain event messages using a simple query
+  mechanism.
+- **Dialer Interface**: Offers a `Dialer` interface for constructing connections,
+  which can be easily mocked for tests.
+- **Observable Pattern**: Integrates the observable pattern, making it easier to
+  react to chain events.
+- **Generic Replay Client**: Offers a generic typed replay client to listen for
+  specifc events on chain, and handles reconnection and subscription on error,
+  if the `EventsQueryClient` returns an error/is closed.
+
+## Usage (`EventsQueryClient`)
```

Comment:
I like how you've added structure to make room here. :sunglasses: 

Wdyt about adding links to the godoc? They will eventually be updated after this is merged. In the meantime you can use the `make go_docs` target to double-check the URL path.